_applications/ceph-source-example.md

```diff
@@ -0,0 +1,88 @@
+---
+title: Ceph Source Example
+link: ceph-source-example
+weight: 100
+labels:
+- Python
+- Ceph
+- S3
+- Jupyter
+layout: application
+menu_template: menu_tutorial_application.html
+menu_items: lightning
+description: |-
+  This is an example of how to connect your application to data in Ceph using S3 API.
+project_links:
+- https://github.com/rimolive/ceph-spark-integration
+---
+<h1 id="introduction">Introduction</h1>
+
+Processing data stored in an external object store is a practical and
+popular way for an intelligent application to operate.
+
+This is an example of the key pieces needed to connect your
+application to data in Ceph using S3 API. It is presented as steps in a Jupyter
+notebook. To run this example, you need ```vagrant``` and ```ansible```.
+
+[Ceph Source Example](https://github.com/radanalyticsio/radanalyticsio.github.io/blob/master/assets/ceph-source-example/ceph-example.ipynb)
+
+[Ceph Ansible playbooks](https://github.com/ceph/ceph-ansible)
+
+<h1 id="architecture">Architecture</h1>
+
+No architecture, this is a connectivity example.
+
+<h1 id="installation">Installation</h1>
+
+Clone [ceph-ansible](https://github.com/ceph/ceph-ansible) GitHub repo.
+
+Rename ```vagrant_variables.yml.sample``` to ```vagrant_variables.yml```
+ and edit the file by changing to the following snippet:
+
+```
+...
+# DEFINE THE NUMBER OF VMS TO RUN
+mon_vms: 1
+osd_vms: 3
+mds_vms: 0
+rgw_vms: 1
+...
+```
+
+Rename also ```site.yml.sample``` to ```site.yml```.
+
+Run ```vagrant up``` inside the repo.
+
+Start a Jupyter notebook with,
+
+```
+oc new-app mattf/workshop:base
+oc expose svc/workshop
+```
+
+Install the boto 3 python library in the notebook pod with,
+
+```
+oc rsh $(oc get pod -l app=workshop | tail -n 1 | awk '{print $1}')
+pip install boto
+```
+
+Find the connection url,
+
+```
+oc logs dc/workshop | grep localhost | sed "s/localhost:8888/$(oc get routes/workshop --template='{% raw %}{{.spec.host}}{% endraw %}')/"
+```
```

Comment:
the newer notebook images allow injection of a password so we don't need to search for the connection string, i think we should leverage that.